tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -0,0 +1,435 @@
+      getSparkProperty(app, "spark.sql.files.maxPartitionBytes") match {
+        case None => bestConfig.set_max_partition_bytes(DEFAULT_MAX_PARTITION_BYTES)
+        case Some(maxPartitionBytes) =>
+          val taskInputSize =
+            app.sqlTaskAggMetrics.map(_.inputBytesReadAvg).sum / app.sqlTaskAggMetrics.size
+          val maxPartitionBytesNum = convertFromHumanReadableSize(maxPartitionBytes)
+          val newMaxPartitionBytes =
+            if (taskInputSize < convertFromHumanReadableSize(MIN_PARTITION_BYTES_RANGE)) {
+              // Increase partition size
+              val calculatedMaxPartitionBytes = Math.max(
+                maxPartitionBytesNum *
+                  (convertFromHumanReadableSize(MIN_PARTITION_BYTES_RANGE) / taskInputSize),
+                convertFromHumanReadableSize(MAX_PARTITION_BYTES_BOUND))
+
+              convertToHumanReadableSize(calculatedMaxPartitionBytes.toLong)
+            } else if (taskInputSize > convertFromHumanReadableSize(MAX_PARTITION_BYTES_RANGE)) {
+              // Decrease partition size
+              val calculatedMaxPartitionBytes = Math.max(
+                maxPartitionBytesNum /
+                  (taskInputSize / convertFromHumanReadableSize(MAX_PARTITION_BYTES_RANGE)),
+                convertFromHumanReadableSize(MAX_PARTITION_BYTES_BOUND))
+
+              convertToHumanReadableSize(calculatedMaxPartitionBytes.toLong)
+            } else {
+              // Do not recommend maxPartitionBytes
+              null
+            }
+
+          bestConfig.set_max_partition_bytes(newMaxPartitionBytes)
+      }
+
+      // Other general recommendations
+      val aqeEnabled = getSparkProperty(app, "spark.sql.adaptive.enabled").getOrElse("False")
+      if (aqeEnabled == "False") {
+        comments :+= "'spark.sql.adaptive.enabled' should be enabled for better performance."
+      }
+
+      // TODO: Check if the fraction calculator is correct.
+      val jvmGCFraction = app.sqlTaskAggMetrics.map {
+        taskMetrics => taskMetrics.jvmGCTimeSum * 1.0 / taskMetrics.executorCpuTime
+      }
+      if ((jvmGCFraction.sum / jvmGCFraction.size) > MAX_JVM_GCTIME_FRACTION) {
+        comments :+= "Average JVM GC time is very high. " +
+          "Other Garbage Collectors can be used for better performance"
+      }
+    }
+  }
+
+  /**
+   * Recommend memory overhead as: pinnedPoolSize + (memoryOverheadFactor * executorMemory)
+   */
+  private def recommendMemoryOverhead(pinnedPoolSize: String, executorMemory: String): Long = {
+    val pinnedPoolSizeNum = convertFromHumanReadableSize(pinnedPoolSize)
+    val executorMemoryNum = convertFromHumanReadableSize(executorMemory)
+    val minMemoryOverhead = convertFromHumanReadableSize(MIN_MEMORY_OVERHEAD)
+    (pinnedPoolSizeNum + Math.max(minMemoryOverhead,
+      DEFAULT_MEMORY_OVERHEAD_FACTOR * executorMemoryNum)).toLong
+  }
+
+  private def recommendGpuProperties(bestConfig: Config, systemProps: SystemProps): Unit = {
```

Comment:
I would call it recommendConfig rather then bestConfig and put description stating what it is